docs/source/tutorial_optimization.rst

```diff
@@ -0,0 +1,900 @@
+ROM model eventually gives less speedup by using a better optimization
+method for the FOM and ROM. This is something that is a common issue for
+MOR methods in the context of optimization.
+
+Breaking the traditional offline/online splitting: enrich along the path of optimization
+----------------------------------------------------------------------------------------
+
+We already figured out that the main drawback for using RB methods in the
+context of optimization is the expensive offline time to build the
+surrogate model. In the example above, we overcame this issue by
+choosing a very high tolerance ``atol``. As a result, we can not be sure
+that our surrogate model is accurate enough for our purpuses. In other
+words, either we invest too much time to build an accurate model or we
+face the danger of reducing with a bad surrogate for the whole parameter
+space. Thinking about this issue again, it is important to notice that
+we are solving an optimization problem which will eventually converge to
+a certain parameter. Thus, it only matters that the surrogate is good in
+this particular region as long as we are able to arrive at it. This
+gives hope that there must exist a more efficient way of using RB
+methods without trying to approximate the FOM across the
+whole parameter space.
+
+One possible way for advanced RB methods is a reduction along the path
+of optimization. The idea is, that we start with an empty basis and only
+enrich the model with the parameters that we will arive at. This
+approach goes beyond the classical offline/online splitting of RB
+methods since it entirely skips the offline phase. In the following
+code, we will test this method.
+
+
+
+.. jupyter-execute::
+
+    pdeopt_reductor = CoerciveRBReductor(
+        fom, product=fom.energy_product, coercivity_estimator=coercivity_estimator)
+
+In the next function, we implement the above mentioned way of enriching
+the basis along the path of optimization.
+
+.. jupyter-execute::
+
+    def record_results_and_enrich(function, data, opt_dict, mu):
+        U = fom.solve(mu)
+        try:
+            pdeopt_reductor.extend_basis(U)
+            data['enrichments'] += 1
+        except:
+            print('Extension failed')
+        opt_rom = pdeopt_reductor.reduce()
+        QoI = rom.output(mu)
+        data['num_evals'] += 1
+        # we need to make sure to copy the data, since the added mu will be changed inplace by minimize afterwards
+        data['evaluation_points'].append([fom.parameters.parse(mu)['diffusion'][:][0], 
+                                          fom.parameters.parse(mu)['diffusion'][:][1]])
+        data['evaluations'].append(QoI[0])
+        opt_dict['opt_rom'] = rom 
+        return QoI
+    
+    def compute_gradient_with_opt_rom(opt_dict, mu):
+        opt_rom = opt_dict['opt_rom']
+        return opt_rom.output_d_mu(opt_rom.parameters.parse(mu), return_array=True, use_adjoint=True)
+
+.. jupyter-execute::
```

Comment:
This seems to be concatenated in the output without any space inbetween. And some text between the cells?